elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -443,9 +453,11 @@ export class PeerPool extends EventEmitter {
 
 		// Throw an error because adding a peer multiple times is a common developer error which is very difficult to identify and debug.
 		if (this._peerMap.has(peer.id)) {
-			throw new Error(`Peer ${peer.id} was already in the peer pool`);
+			this._peerMapDuplicates.set(peer.id, peer);
```

Comment:
I am not sure if necessary but should check to see if it's also already in `_peerMapDuplicates` and throw if so?